hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestNSSummaryEndpointWithOBS.java

```diff
@@ -0,0 +1,1133 @@
+  }
+
+  @Test
+  public void testGetBasicInfoBucketTwo() throws Exception {
+    // Test bucket 2's basics
+    commonUtils.testNSSummaryBasicInfoBucketTwo(
+        BucketLayout.OBJECT_STORE,
+        nsSummaryEndpoint);
+  }
+
+  @Test
+  public void testGetBasicInfoNoPath() throws Exception {
+    // Test invalid path
+    commonUtils.testNSSummaryBasicInfoNoPath(nsSummaryEndpoint);
+  }
+
+  @Test
+  public void testGetBasicInfoKey() throws Exception {
+    // Test key
+    commonUtils.testNSSummaryBasicInfoKey(nsSummaryEndpoint);
+  }
+
+  @Test
+  public void testDiskUsageRoot() throws Exception {
+    // root level DU
+    Response rootResponse = nsSummaryEndpoint.getDiskUsage(ROOT_PATH,
+        false, false);
+    DUResponse duRootRes = (DUResponse) rootResponse.getEntity();
+    assertEquals(2, duRootRes.getCount());
+    List<DUResponse.DiskUsage> duRootData = duRootRes.getDuData();
+    // sort based on subpath
+    Collections.sort(duRootData,
+        Comparator.comparing(DUResponse.DiskUsage::getSubpath));
+    DUResponse.DiskUsage duVol1 = duRootData.get(0);
+    DUResponse.DiskUsage duVol2 = duRootData.get(1);
+    assertEquals(VOL_PATH, duVol1.getSubpath());
+    assertEquals(VOL_TWO_PATH, duVol2.getSubpath());
+    assertEquals(VOL_DATA_SIZE, duVol1.getSize());
+    assertEquals(VOL_TWO_DATA_SIZE, duVol2.getSize());
+  }
+
+  @Test
+  public void testDiskUsageVolume() throws Exception {
+    // volume level DU
+    Response volResponse = nsSummaryEndpoint.getDiskUsage(VOL_PATH,
+        false, false);
+    DUResponse duVolRes = (DUResponse) volResponse.getEntity();
+    assertEquals(2, duVolRes.getCount());
+    List<DUResponse.DiskUsage> duData = duVolRes.getDuData();
+    // sort based on subpath
+    Collections.sort(duData,
+        Comparator.comparing(DUResponse.DiskUsage::getSubpath));
+    DUResponse.DiskUsage duBucket1 = duData.get(0);
+    DUResponse.DiskUsage duBucket2 = duData.get(1);
+    assertEquals(BUCKET_ONE_PATH, duBucket1.getSubpath());
+    assertEquals(BUCKET_TWO_PATH, duBucket2.getSubpath());
+    assertEquals(BUCKET_ONE_DATA_SIZE, duBucket1.getSize());
+    assertEquals(BUCKET_TWO_DATA_SIZE, duBucket2.getSize());
+  }
+
+  @Test
+  public void testDiskUsageBucket() throws Exception {
+    // bucket level DU
+    Response bucketResponse = nsSummaryEndpoint.getDiskUsage(BUCKET_ONE_PATH,
+        false, false);
+    DUResponse duBucketResponse = (DUResponse) bucketResponse.getEntity();
+    // There are no sub-paths under this OBS bucket.
+    assertEquals(0, duBucketResponse.getCount());
+    assertEquals(BUCKET_ONE_DATA_SIZE, duBucketResponse.getSize());
+  }
+
+  @Test
+  public void testDiskUsageKey() throws Exception {
+    // key level DU
+    Response keyResponse = nsSummaryEndpoint.getDiskUsage(KEY_PATH,
+        false, false);
+    DUResponse keyObj = (DUResponse) keyResponse.getEntity();
+    assertEquals(0, keyObj.getCount());
+    assertEquals(KEY_FOUR_SIZE, keyObj.getSize());
+  }
```

Comment:
What's the difference between this and the `testGetBasicInfoKey()` test? Also the `testGetBasicInfoKey()` from the `CommonUtils` seems hard to use in other test classes, or at least we should create the exact same key with the same size, replication, ... to be able to run the test successfully.

Reply:
`testGetBasicInfoKey` triggers the `/summary` endpoint whereas `testDiskUsageKey` over here triggers the `/du` endpoint, which display completely different data.